server/src/main/java/io/unitycatalog/server/persist/ModelRepository.java

```diff
@@ -0,0 +1,361 @@
+        registeredModelInfo.getSchemaName(),
+        registeredModelInfo.getName());
+  }
+
+  private String getRegisteredModelFullName(
+      String catalogName, String schemaName, String modelName) {
+    return catalogName + "." + schemaName + "." + modelName;
+  }
+
+  public UUID getSchemaId(Session session, String catalogName, String schemaName) {
+    SchemaInfoDAO schemaInfo = SCHEMA_REPOSITORY.getSchemaDAO(session, catalogName, schemaName);
+    if (schemaInfo == null) {
+      throw new BaseException(ErrorCode.NOT_FOUND, "Schema not found: " + schemaName);
+    }
+    return schemaInfo.getId();
+  }
+
+  public ListRegisteredModelsResponse listRegisteredModels(
+      String catalogName,
+      String schemaName,
+      Optional<Integer> maxResults,
+      Optional<String> pageToken) {
+    LOGGER.info("Listing registered models in " + catalogName + "." + schemaName);
+    try (Session session = SESSION_FACTORY.openSession()) {
+      session.setDefaultReadOnly(true);
+      Transaction tx = session.beginTransaction();
+      try {
+        UUID schemaId = getSchemaId(session, catalogName, schemaName);
+        ListRegisteredModelsResponse response =
+            listRegisteredModels(session, schemaId, catalogName, schemaName, maxResults, pageToken);
+        tx.commit();
+        return response;
+      } catch (Exception e) {
+        if (tx != null && tx.getStatus().canRollback()) {
+          tx.rollback();
+        }
+        throw e;
+      }
+    }
+  }
+
+  public ListRegisteredModelsResponse listRegisteredModels(
+      Session session,
+      UUID schemaId,
+      String catalogName,
+      String schemaName,
+      Optional<Integer> maxResults,
+      Optional<String> pageToken) {
+    List<RegisteredModelInfoDAO> registeredModelInfoDAOList =
+        LISTING_HELPER.listEntity(session, maxResults, pageToken, schemaId);
+    String nextPageToken = LISTING_HELPER.getNextPageToken(registeredModelInfoDAOList, maxResults);
+    List<RegisteredModelInfo> result = new ArrayList<>();
+    for (RegisteredModelInfoDAO registeredModelInfoDAO : registeredModelInfoDAOList) {
+      RegisteredModelInfo registeredModelInfo = registeredModelInfoDAO.toRegisteredModelInfo();
+      registeredModelInfo.setCatalogName(catalogName);
+      registeredModelInfo.setSchemaName(schemaName);
+      registeredModelInfo.setFullName(getRegisteredModelFullName(registeredModelInfo));
+      result.add(registeredModelInfo);
+    }
+    return new ListRegisteredModelsResponse().registeredModels(result).nextPageToken(nextPageToken);
+  }
+
+  public RegisteredModelInfo updateRegisteredModel(UpdateRegisteredModel updateRegisteredModel) {
+    if (updateRegisteredModel.getNewName() != null) {
+      ValidationUtils.validateSqlObjectName(updateRegisteredModel.getNewName());
+    }
+    if (updateRegisteredModel.getFullName() == null) {
+      throw new BaseException(ErrorCode.INVALID_ARGUMENT, "No model specified.");
```

Comment:
Better error message ?

Reply:
What would be a better message here?  `No three tier full name specified.`?